### ISSUE TITLE
Upgrade failing 7.0.5 to 8.0.1 when keeping apps folder

### ISSUE THREAD

**Issue by jnfrmarks** [owncloud/core#14677]
### Steps to reproduce

1. Install owncloud 7.0.5 community
2. Add a contact to the contacts app
3. upgrade to 8.0.1 community

### Expected behaviour

Community upgrades successfully

### Actual behaviour

Get the white screen after upgrade

Server Log:
```
{
    "app": "PHP",
    "message": "touch(): Unable to create file /var/www/html/owncloud-ce-705/data/owncloud.log because No such file or directory at /var/www/html/owncloud-ce-705/lib/private/log/owncloud.php#49",
    "level": 3,
    "time": "2015-03-03T17:59:36+00:00"
}
```
```
{
    "reqId": "4595786090212ce600927d1fa086f7fa",
    "remoteAddr": "192.168.1.219",
    "app": "index",
    "message": {
        "Exception": {
            "Message": "",
            "Code": 0,
            "Trace": {
                "#0": [
                    "/var/www/html/owncloud-ce-705/lib/private/app.php(77):",
                    "OC_App::loadApp('activity')"
                ],
                "#1": [
                    "/var/www/html/owncloud-ce-705/lib/private/util.php(75):",
                    "OC_App::loadApps(Array)"
                ],
                "#2": [
                    "/var/www/html/owncloud-ce-705/lib/base.php(766):",
                    "OC_Util::setupFS()"
                ],
                "#3": [
                    "/var/www/html/owncloud-ce-705/index.php(36):",
                    "OC::handleRequest()"
                ],
                "#4": [
                    "{main}",
                    null
                ]
            },
            "File": "/var/www/html/owncloud-ce-705/lib/private/app.php",
            "Line": 96
        }
    },
    "level": 4,
    "time": "2015-03-03T18:02:39+00:00"
}
```
```
{
    "reqId": "eac16da71dac56d3be71ef49dac94520",
    "remoteAddr": "192.168.1.219",
    "app": "PHP",
    "message": "Class OCA\Contacts\Share\Addressbook contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (OCP\Share_Backend::isShareTypeAllowed) at /var/www/html/owncloud-ce-705/apps/contacts/lib/share/addressbook.php#130",
    "level": 3,
    "time": "2015-03-03T18:03:45+00:00"
}
```

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration

ownCloud Edition 8.0.1 (stable)

**Comment by jnfrmarks** [comment]
@PVince81 @karlitschek

**Comment by PVince81** [comment]
Hmmm, I thought we were disabling 3rd party apps now during upgrade ? @DeepDiver1975

**Comment by PVince81** [comment]
@jnfrmarks I assume you are talking about tarball based updates (not RPM or DEB)

**Comment by jnfrmarks** [comment]
Yes, I'm testing with tarballs.  In this test case, the contacts apps was not detected as third party.

**Comment by karlitschek** [comment]
@PVince81 @DeepDiver1975 Not sure how we detect "3rdparty" apps. Maybe the contacts app wasn´t detected correctly as 3rdparty because it was a shipped app in 7.

**Comment by PVince81** [comment]
Yeah, very likely: https://github.com/owncloud/core/pull/14282/files#diff-499598b2af703b899a7981be42dec8e0R376

But also comes again the question of the upgrade process.
@jnfrmarks did you do the "delete everything except data, config, extract new tarball" process ?
In that case "everything" should include the "apps" folder that contains contacts, so it should be gone.

**Comment by jnfrmarks** [comment]
I did all of the above and THEN copied in the contacts folder from 7.0.5.

I am getting much better behavior when copying over the 8.0.1 contacts folder from the app store - we properly come up, the contacts app is detected as third party, and is not enabled - all expected.

I'm doing some additional test cases as requested by @MTRichards and @karlitschek

**Comment by jnfrmarks** [comment]
ok - so here's what I have found out:

If before I start the upgrade to 8, I copy over the latest version of contacts from the appstore into the apps directory, things works perfectly: the app is detected as third party and not enabled.  Once I enable it, it displays the one contact I created prior to the upgrade.

**Comment by PVince81** [comment]
Hmmm, I think you're not supposed to copy over the old contacts app from OC 7 as it is not compatible.
But I'm pretty sure some users would be tempted to do so. (and ignore the documentation)
That contacts/calendar apps should actually be reinstalled from the apps page directly (they will be downloaded automatically).

**Comment by MTRichards** [comment]
>That contacts/calendar apps should actually be reinstalled from the apps page directly (they will be downloaded automatically).

Is that supposed to happen automatically, or does it require me to push a button?

**Comment by PVince81** [comment]
Currently it requires you to go to the apps page and manually push a button for every 3rd party app you want to reinstall. I agree that it is not ideal from the UX point of view.

In upgrade cases where 3rd party apps were automatically disabled the same manipulation is needed.

**Comment by DeepDiver1975** [comment]
> Maybe the contacts app wasn´t detected correctly as 3rdparty because it was a shipped app in 7.

this is exactly what happens - oc7 contacts and calendar apps have been marked as 'shipped' and therefore will not be detected.

for sure not nice - we can try to enhance this for future releases - question is if this will ever happen again?

**Comment by nickvergessen** [comment]
We could hardcode the check to disable calendar and contacts aswell?

**Comment by DeepDiver1975** [comment]
> We could hardcode the check to disable calendar and contacts aswell?

sure - but the question is if this will help - any change will be part of 8.0.2 - from my pov all instances did already fail while upgrading to 8.0.0 or 8.0.1

**Comment by DeepDiver1975** [comment]
I guess we can close this issue - has been addressed in #14711
